src/Microsoft.Azure.SignalR.AspNet/Middleware/NegotiateMiddleware.cs

```diff
@@ -184,10 +184,10 @@ private Task ProcessNegotiationRequest(IOwinContext owinContext, HostContext con
             {
                 return GenerateClientAccessTokenAsync(provider, context, url, claims);
             }
-            catch (AzureSignalRAccessTokenNotAuthorizedException e)
+            catch (AzureSignalRCredentialUnauthorizedException e)
             {
                 Log.NegotiateFailed(_logger, e.Message);
-                context.Response.StatusCode = 500;
+                context.Response.StatusCode = 403;
```

Comment:
it sounds 500 since it is the app server having no permission not the client?

Reply:
sounds correct.